simulator/func_sim/alu.h

```diff
@@ -158,27 +158,54 @@ struct ALU
     template<typename I, typename T> static void riscv_subtraction( I* instr)  { instr->v_dst = sign_extension<bitwidth<T>, typename I::RegisterUInt>(narrow_cast<T>( instr->v_src1) - narrow_cast<T>( instr->v_src2)); }
     template<typename I, typename T> static void addition_imm( I* instr) { instr->v_dst = narrow_cast<T>( instr->v_src1) + narrow_cast<T>( instr->v_imm); }
 
+    template<typename T, typename T_src1, typename T_src2> static
+    bool is_addition_overflow( T_src1 src1, T_src2 src2)
+    {
+        if ( std::numeric_limits<T>::max() - narrow_cast<T>( src1) < narrow_cast<T>( src2))
+            return true;
+        return false;
```

Comment:
```c++
return  std::numeric_limits<T>::max() - narrow_cast<T>( src1) < narrow_cast<T>( src2);
```

Reply:
Yes, this is wrong. I work with unsigned overflow, but need two's complement.